osu.Game/Storyboards/Drawables/DrawableStoryboardSprite.cs

```diff
@@ -84,24 +90,28 @@ public DrawableStoryboardSprite(StoryboardSprite sprite)
             LifetimeEnd = sprite.EndTimeForDisplay;
         }
 
-        [Resolved]
-        private ISkinSource skin { get; set; } = null!;
-
         [BackgroundDependencyLoader]
-        private void load(TextureStore textureStore, Storyboard storyboard)
+        private void load(Storyboard storyboard)
         {
-            Texture = textureStore.Get(Sprite.Path);
-
-            if (Texture == null && storyboard.UseSkinSprites)
+            if (storyboard.UseSkinSprites)
             {
                 skin.SourceChanged += skinSourceChanged;
                 skinSourceChanged();
             }
+            else
+                Texture = textureStore.Get(Sprite.Path);
 
             Sprite.ApplyTransforms(this);
         }
 
-        private void skinSourceChanged() => Texture = skin.GetTexture(Sprite.Path);
+        private void skinSourceChanged()
+        {
+            // Setting texture will only update the size if it's zero.
+            // So let's force an update by setting to zero.
+            Size = Vector2.Zero;
```

Comment:
This will fail [if this early guard](https://github.com/ppy/osu-framework/blob/ff3829952c3a39b7bcf637b531d840f3df6ab2a9/osu.Framework/Graphics/Sprites/Sprite.cs#L154-L155) is hit, which I've hit in practice.

Suggest applying

```diff
diff --git a/osu.Game/Storyboards/Drawables/DrawableStoryboardSprite.cs b/osu.Game/Storyboards/Drawables/DrawableStoryboardSprite.cs
index 87c09c3e5e..68892a6fc1 100644
--- a/osu.Game/Storyboards/Drawables/DrawableStoryboardSprite.cs
+++ b/osu.Game/Storyboards/Drawables/DrawableStoryboardSprite.cs
@@ -106,11 +106,11 @@ private void load(Storyboard storyboard)
 
         private void skinSourceChanged()
         {
-            // Setting texture will only update the size if it's zero.
-            // So let's force an update by setting to zero.
-            Size = Vector2.Zero;
-
             Texture = skin.GetTexture(Sprite.Path) ?? textureStore.Get(Sprite.Path);
+
+            // Setting texture will only update the size if it's zero.
+            // So let's force an explicit update.
+            Size = new Vector2(Texture.DisplayWidth, Texture.DisplayHeight);
         }
 
         protected override void Dispose(bool isDisposing)

```

instead.

Reply:
Is what I originally had. Sounds fine.